### PR TITLE
Release V1

### PR DESCRIPTION
**Pull request by fracek**
The goal of this PR is to prepare for the v1 release, which brings Starknet.js v5 compatibility.

### CODE REVIEW

**Comment by fracek** [comment]
Need to add a changelog before we can release

**Comment by fracek** [comment]
Hey @dhruvkelawala , if you have time to give this a quick look it would be great!

**Comment by dhruvkelawala** [comment]
Will review it tomorrow! Thanks!

File: packages/core/src/hooks/receipt.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { useMemo } from 'react'
 import { GetTransactionReceiptResponse, ProviderInterface } from 'starknet'
 import { useStarknet } from '../providers'
 import { useInvalidateOnBlock } from './invalidate'
+import { Chain, useNetwork } from '..'
```
Comment (dhruvkelawala) [inline packages/core/src/hooks/receipt.ts:6]:
```suggestion
import { Chain, useNetwork } from './network'
```

File: packages/core/src/hooks/transaction.ts [hunk 2]
```diff
@@ -2,6 +2,8 @@ import { useQueries, useQuery } from '@tanstack/react-query'
 import { GetTransactionResponse, ProviderInterface } from 'starknet'
 import { useStarknet } from '../providers'
 import { useInvalidateOnBlock } from './invalidate'
+import { useNetwork } from './network'
+import { Chain } from '..'
```
Comment (dhruvkelawala) [inline packages/core/src/hooks/transaction.ts:6]:
```suggestion
import { useNetwork, Chain } from './network'
```

File: packages/core/src/providers/starknet.tsx [hunk 3]
```diff
@@ -39,7 +31,7 @@ export interface StarknetState {
 // as defaultProvider from starknet.js is initialized to testnet2
 const customDefaultProvider = new Provider({
   sequencer: {
-    network: 'goerli-alpha',
+    network: constants.NetworkName.SN_GOERLI,
```
Comment (dhruvkelawala) [inline packages/core/src/providers/starknet.tsx:34]: Let's default to mainnet as we discussed?

File: packages/core/src/providers/starknet.tsx [hunk 4]
```diff
@@ -209,7 +201,7 @@ function useStarknetManager({
       }
 
       const lastConnectedConnector = connectors.find(
-        (connector) => connector.id() === lastConnectedConnectorId
+        (connector) => connector.id === lastConnectedConnectorId
```
Comment (dhruvkelawala) [inline packages/core/src/providers/starknet.tsx:204]: This looks much better

**Review by dhruvkelawala — CHANGES_REQUESTED** [review]

**Comment by dhruvkelawala** [comment]
Just some minor changes required, otherwise lgtm!

**Comment by fracek** [comment]
Thank you! Tomorrow I'll double check everything and release this.

**Pull request merged**